net/nimble/controller/src/ble_ll_test.c

```diff
@@ -0,0 +1,498 @@
+                0x3c, 0xbe, 0x0b, 0xd6, 0x76, 0x83, 0xd6, 0x57,
+                0x05, 0x4a, 0x3d, 0xdd, 0x81, 0x73, 0xc9, 0xeb,
+                0x8a, 0x84, 0x39, 0xf4, 0x36, 0x0b, 0xf7 };
+
+static const uint8_t g_ble_ll_test_prbs15_data[] = {
+                0xff, 0x7f, 0xf0, 0x3e, 0x3a, 0x13, 0xa4, 0xdc,
+                0xe2, 0xf9, 0x6c, 0x54, 0xe2, 0xd8, 0xea, 0xc8,
+                0x88, 0x00, 0x21, 0x86, 0x9c, 0x6a, 0xd8, 0xcb,
+                0x4e, 0x14, 0x6a, 0xf9, 0x4d, 0xd2, 0x7e, 0xb2,
+                0x32, 0x03, 0xc6, 0x14, 0x4b, 0x7f, 0xd1, 0xb8,
+                0xa6, 0x79, 0x7c, 0x17, 0xac, 0xed, 0x06, 0xad,
+                0xaf, 0x0a, 0x94, 0x7a, 0xba, 0x03, 0xe7, 0x92,
+                0xd7, 0x15, 0x09, 0x73, 0xe8, 0x6d, 0x16, 0xee,
+                0xe1, 0x3f, 0x78, 0x1f, 0x9d, 0x09, 0x52, 0x6e,
+                0xf1, 0x7c, 0x36, 0x2a, 0x71, 0x6c, 0x75, 0x64,
+                0x44, 0x80, 0x10, 0x43, 0x4e, 0x35, 0xec, 0x65,
+                0x27, 0x0a, 0xb5, 0xfc, 0x26, 0x69, 0x3f, 0x59,
+                0x99, 0x01, 0x63, 0x8a, 0xa5, 0xbf, 0x68, 0x5c,
+                0xd3, 0x3c, 0xbe, 0x0b, 0xd6, 0x76, 0x83, 0xd6,
+                0x57, 0x05, 0x4a, 0x3d, 0xdd, 0x81, 0x73, 0xc9,
+                0xeb, 0x8a, 0x84, 0x39, 0xf4, 0x36, 0x0b, 0xf7,
+                0xf0, 0x1f, 0xbc, 0x8f, 0xce, 0x04, 0x29, 0xb7,
+                0x78, 0x3e, 0x1b, 0x95, 0x38, 0xb6, 0x3a, 0x32,
+                0x22, 0x40, 0x88, 0x21, 0xa7, 0x1a, 0xf6, 0xb2,
+                0x13, 0x85, 0x5a, 0x7e, 0x93, 0xb4, 0x9f, 0xac,
+                0xcc, 0x80, 0x31, 0xc5, 0xd2, 0x5f, 0x34, 0xae,
+                0x69, 0x1e, 0xdf, 0x05, 0x6b, 0xbb, 0x41, 0xeb,
+                0xab, 0x02, 0xa5, 0x9e, 0xee, 0xc0, 0xb9, 0xe4,
+                0x75, 0x45, 0xc2, 0x1c, 0x7a, 0x9b, 0x85, 0x7b,
+                0xf8, 0x0f, 0xde, 0x47, 0x67, 0x82, 0x94, 0x5b,
+                0x3c, 0x9f, 0x8d, 0x4a, 0x1c, 0x5b, 0x1d, 0x19,
+                0x11, 0x20, 0xc4, 0x90, 0x53, 0x0d, 0x7b, 0xd9,
+                0x89, 0x42, 0x2d, 0xbf, 0x49, 0xda, 0x4f, 0x56,
+                0x66, 0xc0, 0x98, 0x62, 0xe9, 0x2f, 0x1a, 0xd7,
+                0x34, 0x8f, 0xef, 0x82, 0xb5, 0xdd, 0xa0, 0xf5,
+                0x55, 0x81, 0x52, 0x4f, 0x77, 0xe0, 0x5c, 0xf2,
+                0xba, 0x22, 0x61, 0x0e, 0xbd, 0xcd, 0xc2 };
+
+#define BLE_TEST_SYNC_WORD          (0x71764129)
+#define BLE_TEST_CRC                (0x555555)
+
+static void
+ble_ll_test_set_next(struct test_ctx *ctx)
+{
+    struct ble_ll_sched_item *sch = &ctx->sch;
+
+    sch->start_time += ctx->itvl_ticks;
+    sch->remainder += ctx->itvl_rem_usec;
+    if (sch->remainder > 30) {
+       sch->start_time++;
+       sch->remainder -= 30;
+    }
+
+    sch->start_time -= g_ble_ll_sched_offset_ticks;
+}
+
+static void
+ble_ll_test_event(struct os_event *evt) {
+    /* It is called in LL context */
+    struct test_ctx *c = evt->ev_arg;
+    int rc;
+    os_sr_t sr;
+
+    OS_ENTER_CRITICAL(sr);
```

Comment:
why you need to handle this event with interrupts disabled? does not seem to be necessary for other scheduler items

Reply:
this is os_event run in LL context. Not the scheduler item